File: decompiled/sources/com/hjq/http/model/HttpParams.java
package com.hjq.http.model;

import com.hjq.http.EasyConfig;
import java.util.HashMap;
import java.util.Set;

public final class HttpParams {
    private boolean mMultipart;
    private HashMap<String, Object> mParams = EasyConfig.getInstance().getParams();

    public void put(String str, Object obj) {
        if (str != null && obj != null) {
            if (this.mParams == EasyConfig.getInstance().getParams()) {
                this.mParams = new HashMap<>(this.mParams);
            }
            this.mParams.put(str, obj);
        }
    }

    public void remove(String str) {
        if (str != null) {
            if (this.mParams == EasyConfig.getInstance().getParams()) {
                this.mParams = new HashMap<>(this.mParams);
            }
            this.mParams.remove(str);
        }
    }

    public Object get(String str) {
        return this.mParams.get(str);
    }

    public boolean isEmpty() {
        HashMap<String, Object> hashMap = this.mParams;
        return hashMap == null || hashMap.isEmpty();
    }

    public Set<String> getNames() {
        return this.mParams.keySet();
    }

    public HashMap<String, Object> getParams() {
        return this.mParams;
    }

    public boolean isMultipart() {
        return this.mMultipart;
    }

    public void setMultipart(boolean z) {
        this.mMultipart = z;
    }
}

File: decompiled/sources/com/hjq/http/model/HttpParams.kt
package com.hjq.http.model

import com.hjq.http.EasyConfig
import java.util.HashMap

class HttpParams {
    private var mMultipart: Boolean = false
    private var mParams: HashMap<String, Any> = EasyConfig.getInstance().params

    fun put(str: String?, obj: Any?) {
        if (str != null && obj != null) {
            if (mParams === EasyConfig.getInstance().params) {
                mParams = HashMap(mParams)
            }
            mParams[str] = obj
        }
    }

    fun remove(str: String?) {
        if (str != null) {
            if (mParams === EasyConfig.getInstance().params) {
                mParams = HashMap(mParams)
            }
            mParams.remove(str)
        }
    }

    fun get(str: String): Any? {
        return mParams[str]
    }

    fun isEmpty(): Boolean {
        return mParams.isEmpty()
    }

    fun getNames(): Set<String> {
        return mParams.keys
    }

    fun getParams(): HashMap<String, Any> {
        return mParams
    }

    fun isMultipart(): Boolean {
        return mMultipart
    }

    fun setMultipart(z: Boolean) {
        mMultipart = z
    }
}
